src/test/java/core/basesyntax/io/FileWriterImplTest.java

```diff
@@ -0,0 +1,66 @@
+package core.basesyntax.io;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class FileWriterImplTest {
+
+    private FileWriterImpl fileWriter;
+    private final String filePath = "testFile.txt";
+
+    @BeforeEach
+    void setUp() {
+        fileWriter = new FileWriterImpl();
+    }
+
+    @Test
+    void shouldWriteDataCorrectly_whenFilePathIsValid() throws IOException {
+        String data = "Hello, FileWriter!";
+        fileWriter.write(data, filePath);
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line = reader.readLine();
+            assertEquals(data, line,
+                    "The content written to the file should match the expected data");
```

Comment:
move check after try block. Also, it is better to use try-catch and move it in a private method